paimon-core/src/main/java/org/apache/paimon/stats/SimpleStatsEvolutions.java

```diff
@@ -78,19 +86,27 @@ public SimpleStatsEvolution getOrCreate(long dataSchemaId) {
     }
 
     @Nullable
-    public Predicate tryDevolveFilter(long dataSchemaId, Predicate filter) {
+    public Predicate tryDevolveFilter(long dataSchemaId, Predicate filter, boolean nullSafe) {
         if (tableSchemaId == dataSchemaId) {
             return filter;
         }
         List<Predicate> devolved =
                 Objects.requireNonNull(
                         SchemaEvolutionUtil.devolveDataFilters(
-                                schemaFields.apply(tableSchemaId),
+                                tableDataFields,
                                 schemaFields.apply(dataSchemaId),
-                                Collections.singletonList(filter)));
+                                Collections.singletonList(filter),
+                                isKeyStats,
+                                nullSafe));
         return devolved.isEmpty() ? null : devolved.get(0);
```

Comment:
Here should consider and filter, should deal with multiple predicates situation.